### ISSUE TITLE
Anything similar to Golang clientcmd ModifyConfig?

### ISSUE THREAD

**Issue by ZedYeung** [kubernetes-client/javascript#340]
It looks like currently, we could only read the config file, is there any method like Golang clientcmd ModifyConfig so that we could modify the config file?

**Comment by brendandburns** [comment]
Not currently. Can you describe your use case in more detail? Do you want to modify and then save to file a kubeconfig?

**Comment by ZedYeung** [comment]
yes, ideally, i hope it could be same as  Golang clientcmd ModifyConfig.

**Comment by brendandburns** [comment]
For future reference, code is here:
https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/config.go#L158

Hrm. Why not just modify the config object in Javascript and then export to YAML?

That function seems super twisty and it's not exactly clear to me why its useful.

Can you describe your use case in more detail?

Thanks

**Comment by ZedYeung** [comment]
That is exactly what it did. It is very true that function is super twisty with some rules. I  just want to keep consistent with that official rule to modify the config file, in case some specific rule would be broken. But looks like that is not easy. What I do now is just modifying only one file in the order `env KUBECONFIG and then ~/.kube/config`. Hope that is good.

**Comment by brendandburns** [comment]
That seems reasonable for now. I'll leave this issue open, but I don't expect we'll implement this in the near term.

**Comment by machine424** [comment]
If I really understand how tokens refreshing is implemented, new fetched tokens (at least for oidc) are not stored in `~/.kube/config`, that means, if a refresh occurs, `~/.kube/config` becomes obsolete. Also if other clients are to be run after, they will get old and unusable tokens from `~/.kube/config`. The current refreshing process is only useful for a single long-running client ? I see that you've started working on configuration exporting (https://github.com/kubernetes-client/javascript/pull/372) Will tokens persistence be implemented soon ? Should it be done by the app using the client ?

**Comment by brendandburns** [comment]
Token persistence is tricky, because it is potentially racy between multiple different clients running on the same machine.

I'd prefer that people handle it themselves in their application code rather than this library accidentally corrupting someone's kubeconfig.
